source/Ship.cpp

```diff
@@ -2501,43 +2500,67 @@ int Ship::Scan()
-	double cargoDistance = 100. * sqrt(attributes.Get("cargo scan power"));
-	double outfitDistance = 100. * sqrt(attributes.Get("outfit scan power"));
+	// Because of Pythagoras, if we use square-distance, we can skip this square root.
+	double cargoDistance = attributes.Get("cargo scan power");
+	double outfitDistance = attributes.Get("outfit scan power");
 
 	// Bail out if this ship has no scanners.
 	if(!cargoDistance && !outfitDistance)
 		return 0;
 
-	// Scanning speed also uses a square root, so you need four scanners to get
-	// twice the speed out of them.
-	double cargoSpeed = sqrt(attributes.Get("cargo scan speed"));
+	// Scanning speed will be reduced by the square-distance to the target, which is
+	// close to above scanner range.
+	double cargoSpeed = attributes.Get("cargo scan speed");
 	if(!cargoSpeed)
 		cargoSpeed = 1.;
-	double outfitSpeed = sqrt(attributes.Get("outfit scan speed"));
+
+	double outfitSpeed = attributes.Get("outfit scan speed");
 	if(!outfitSpeed)
 		outfitSpeed = 1.;
 
 	// Check how close this ship is to the target it is trying to scan.
-	double distance = (target->position - position).Length();
+	// To normalize 1 "scan power" to reach 100 pixels, divide this square distance by 100^2, or multiply by 0.0001.
+	// Because we use distance squared, to reach 200 pixels away you need 4 "scan power".
+	double distance = (target->position - position).LengthSquared() * .0001;
+
+	// Check the target's outfit and cargo space. A larger ship takes longer to scan.
+	// Normalized around 200 tons of cargo/outfit space.
+	double outfits = target->baseAttributes.Get("outfit space") * .005;
+	double cargo = target->attributes.Get("cargo space") * .005;
 
 	// Check if either scanner has finished scanning.
 	bool startedScanning = false;
 	bool activeScanning = false;
 	int result = 0;
-	auto doScan = [&](double &elapsed, const double speed, const double scannerRange, const int event) -> void
+	auto doScan = [&](double &elapsed, const double speed, const double scannerRange, const double depth, const int event) -> void
 	{
 		if(elapsed < SCAN_TIME && distance < scannerRange)
 		{
 			startedScanning |= !elapsed;
 			activeScanning = true;
-			// To make up for the scan decay above:
-			elapsed += speed + 1.;
+
+			/* Division is more expensive to calculate than multiplication,
+			so rearrange the formula to minimize divisions.
+
+			"(scannerRange - distance) / scannerRange"
+			This line hits 1 at distace = 0, and 0 at distance = scannerRange.
+			This is a hard cap on scanning range.
+
+			"speed / (sqrt(speed) + distance)"
+			This gives a modest speed boost at no distance (speed/sqrt(speed), and
+			the boost tapers off to 0 at arbitrary distance.
+
+			"1 / depth"
+			This doubles scan time for cargo holds or outfit sections that are twice as large. */
+			elapsed += ((scannerRange - distance) * speed) / (scannerRange * (sqrt(speed) + distance) * depth);
+			// To make up for the scan decay:
+			++elapsed;
```

Comment:
I'd just combine these two lines.
```suggestion
			// The final +1 is to make up for scan decay that occurred this frame.
			elapsed += ((scannerRange - distance) * speed) / (scannerRange * (sqrt(speed) + distance) * depth) + 1;
```